src/main/java/org/opensearch/timeseries/util/RestHandlerUtils.java

```diff
@@ -130,18 +135,18 @@ public static XContentParser createXContentParserFromRegistry(NamedXContentRegis
     }
 
     /**
-     * Check if there is configuration/syntax error in feature definition of anomalyDetector
-     * @param anomalyDetector detector to check
-     * @param maxAnomalyFeatures max allowed feature number
+     * Check if there is configuration/syntax error in feature definition of config
+     * @param config config to check
+     * @param maxFeatures max allowed feature number
      * @return error message if error exists; otherwise, null is returned
      */
-    public static String checkAnomalyDetectorFeaturesSyntax(AnomalyDetector anomalyDetector, int maxAnomalyFeatures) {
-        List<Feature> features = anomalyDetector.getFeatureAttributes();
+    public static String checkFeaturesSyntax(Config config, int maxFeatures) {
+        List<Feature> features = config.getFeatureAttributes();
         if (features != null) {
-            if (features.size() > maxAnomalyFeatures) {
-                return "Can't create more than " + maxAnomalyFeatures + " anomaly features";
+            if (features.size() > maxFeatures) {
+                return "Can't create more than " + maxFeatures + " features";
             }
-            return validateFeaturesConfig(anomalyDetector.getFeatureAttributes());
+            return validateFeaturesConfig(config.getFeatureAttributes());
         }
         return null;
```

Comment:
Also in the method `validateFeaturesConfig` in this file we have this line: `errorMsgBuilder.append("Detector has duplicate feature names: ");` We should probably change to either "There are duplicate feature names"/"Given configurations have duplicate feature names" or exact the class being used (Forecaster/detector) for the given messaging. In short term wont really be an issue since max feature number will be one for forecasting